PowerShell_Scripts/FolderSynchronisation_edited+func_2ed.ps1

```diff
@@ -0,0 +1,27 @@
+﻿Clear-Host
+
+Write-Host "We will ask you to indicate the path for folder 1 and folder 2"
+function Path($TestFolderPath) {
+    Write-Host "Please, specify the path"
+    $TestFolderPath = Read-Host
+
+    return $TestFolderPath
+}
+
+$TestFolder1Path = Path #'C:\Users\Daria\Desktop\TestFolder1'
+$TestFolder2Path = Path #'C:\Users\Daria\Desktop\TestFolder2'
+
+$Folder1Content = Get-ChildItem $TestFolder1Path -Recurse
+$Folder2Content = Get-ChildItem $TestFolder2Path -Recurse
+
+$Difference1 = Compare-Object -ReferenceObject $Folder1Content -DifferenceObject $Folder2Content `
+-Property Name, Length -PassThru | Where-Object {$_.SideIndicator -eq "<="}
```

Comment:
This is a very interesting approach to the problem. It makes a great use of existing PS capabilities however it also sacrifices readability and may be hard to alter if we ever decide to change the way we compare files. For this reason that approach, while being sound for a one-off script, can be less than ideal for big scripts.

Of course there's more than one way to bake a cake but if you want few pointers, I recommend going through the files in a loop and calling a new comparison function that will receive both files as arguments and the body of which will contain the comparison logic.

Reply:
May be this variant will be more appropriate?

```suggestion
function CompareContent {
    param ($folderpath, $foldercontent1, $foldercontent2)

    If ($foldercontent1 -ne $foldercontent2 ) {
        Copy-Item $foldercontent1 -Destination $folderpath -Force
    }

    return $folderpath, $foldercontent1, $foldercontent2
}

CompareContent $TestFolder2Path $Folder1Content $Folder2Content
CompareContent $TestFolder1Path $Folder2Content $Folder1Content
```